test/Microsoft.SqlTools.ServiceLayer.UnitTests/ObjectExplorer/NodeFilterTests.cs

```diff
@@ -0,0 +1,195 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.ServiceLayer.ObjectExplorer;
+using Microsoft.SqlTools.ServiceLayer.ObjectExplorer.Nodes;
+using Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel;
+using NUnit.Framework;
+
+namespace Microsoft.SqlTools.ServiceLayer.UnitTests.ObjectExplorer
+{
+    public class NodeFilterTests
+    {
+
+        // Basic PropertyFilter definitions to use in tests
+        public NodePropertyFilter TemporalFilter = 
+            new NodePropertyFilter
+                {
+                    Property = "TemporalType",
+                    Type = typeof(Enum),
+                    TypeToReverse = typeof(SqlHistoryTableQuerier),
+                    ValidFor = ValidForFlag.Sql2016|ValidForFlag.Sql2017|ValidForFlag.Sql2019|ValidForFlag.Sql2022|ValidForFlag.AzureV12,
+                    Values = new List<object>
+                    {
+                        { TableTemporalType.HistoryTable }
+                    }
+                };
+
+        public NodePropertyFilter LedgerHistoryFilter = 
+            new NodePropertyFilter
+                {
+                    Property = "LedgerType",
+                    Type = typeof(Enum),
+                    TypeToReverse = typeof(SqlHistoryTableQuerier),
+                    ValidFor = ValidForFlag.Sql2022|ValidForFlag.AzureV12,
+                    Values = new List<object>
+                    {
+                        { LedgerTableType.HistoryTable }
+                    }
+                };
+
+        public NodePropertyFilter SystemObjectFilter =
+            new NodePropertyFilter
+                {
+                    Property = "IsSystemObject",
+                    Type = typeof(bool),
+                    Values = new List<object> { 1 },
+                };
+
+        /// <summary>
+        /// Validates the output of the ToPropertyFilterString for the NodeOrFilter class
+        /// </summary>
+        [Test]
+        public void NodeOrFilterReturnsProperString()
+        {
+            var orNode = new NodeOrFilter {
+                FilterList = new List<NodePropertyFilter> {
+                    TemporalFilter,
+                    LedgerHistoryFilter
+                }
+            };
+
+            string validForBoth = orNode.ToPropertyFilterString(typeof(SqlHistoryTableQuerier), ValidForFlag.Sql2022);
+            string expectedBothFilters = "((@TemporalType = 1) or (@LedgerType = 1))";
+            Assert.AreEqual(expectedBothFilters, validForBoth);
```

Comment:
Asserts should use NUnit fluent assertions (`Assert.That(validForBoth, Is.EqualTo(expectedBothFilters))`) and then also generally include a message indicating what exactly failed (especially when there's multiple asserts in a test).

Reply:
Updated accordingly. Previously I was just following coding standards for the rest of the ObjectExplorer unit tests.

Reply:
Yeah, unfortunately most of the original tests were written with the base assert methods. The fluent assertions are a lot easier to read and provide better failure messages so we've moved to using those for newer tests but haven't had time to go back and update all the older tests yet.